src/mock_sim.cc

```diff
@@ -0,0 +1,159 @@
+#include "mock_sim.h"
+
+#include "cyclus.h"
+#include <sstream>
+
+namespace cyclus {
+
+// The code for this function was copied with minor adjustements from
+// XMLFileLoader::LoadInitialAgents
+void InitAgent(Agent* a, std::stringstream& config, Recorder* rec,
+               SqliteBack* back) {
+  XMLParser parser_;
+  parser_.Init(config);
+  InfileTree xqe(parser_);
+
+  a->Agent::InfileToDb(&xqe, DbInit(a, true));
+  a->InfileToDb(&xqe, DbInit(a));
+  rec->Flush();
+
+  std::vector<Cond> conds;
+  conds.push_back(Cond("SimId", "==", rec->sim_id()));
+  conds.push_back(Cond("SimTime", "==", static_cast<int>(0)));
+  conds.push_back(Cond("AgentId", "==", a->id()));
+  CondInjector ci(back, conds);
+
+  // call manually without agent impl injected
+  PrefixInjector pi(&ci, "AgentState");
+  a->Agent::InitFrom(&pi);
+
+  pi = PrefixInjector(&ci, "AgentState" + AgentSpec(a->spec()).Sanitize());
+  a->InitFrom(&pi);
+}
+
+///////// MockAgent ////////////
+
+int MockAgent::nextid_ = 0;
+
+MockAgent::MockAgent(Context* ctx, Recorder* rec, SqliteBack* b, bool is_source)
+    : ctx_(ctx),
+      rec_(rec),
+      back_(b),
+      source_(is_source),
+      cap_(1e299),
+      lifetime_(-1),
+      start_(0) {
+  std::stringstream ss;
+  if (is_source) {
+    ss << "source";
+  } else {
+    ss << "sink";
+  }
+  ss << nextid_++;
+  proto_ = ss.str();
+}
+
+MockAgent MockAgent::commod(std::string commod) {
+  commod_ = commod;
+  return *this;
+}
+MockAgent MockAgent::recipe(std::string recipe) {
+  recipe_ = recipe;
+  return *this;
+}
+MockAgent MockAgent::capacity(double cap) {
+  cap_ = cap;
+  return *this;
+}
+MockAgent MockAgent::start(int t) {
+  start_ = t;
+  return *this;
+}
+MockAgent MockAgent::lifetime(int duration) {
+  lifetime_ = duration;
+  return *this;
+}
+
+std::string MockAgent::Finalize() {
+  AgentSpec spec(":agents:Sink");
+  if (source_) {
+    spec = AgentSpec(":agents:Source");
+  }
+
+  std::stringstream xml;
+  xml << "<facility><name>" << proto_ << "</name><lifetime>" << lifetime_
+      << "</lifetime><config><SrcSnkAgent>";
+
+  if (source_) {
+    xml << "<commod>" << commod_ << "</commod>";
+  } else {
+    xml << "<in_commods><val>" << commod_ << "</val></in_commods>";
+  }
+  xml << "<capacity>" << cap_ << "</capacity>";
+  if (recipe_ != "") {
+    xml << "<recipe_name>" << recipe_ << "</recipe_name>";
+  }
+  xml << "</SrcSnkAgent></config></facility>";
+
+  Agent* a = DynamicModule::Make(ctx_, spec);
+  InitAgent(a, xml, rec_, back_);
+
+  ctx_->AddPrototype(proto_, a);
+
+  if (start_ == 0) {
+    a = ctx_->CreateAgent<Agent>(proto_);
+    a->Build(NULL);
+  } else {
+    ctx_->SchedBuild(NULL, proto_, start_);
+  }
+  return proto_;
+}
+
+///////// MockSim ////////////
+
+MockSim::MockSim(AgentSpec spec, std::string config, int duration)
+    : ctx_(&ti_, &rec_), back_(NULL) {
+  back_ = new SqliteBack(":memory:");
```

Comment:
To clarify - I do no special handling of that keyword as a filepath - sqlite does it all internally - so it is not under my control.